pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,275 @@
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+//! # Rewards Module
+//!
+//! TODO.
+//!
+//! ## Overview
+//!
+//! TODO
+//!
+//! ## Interface
+//!
+//! ### Dispatchable Functions
+//!
+//! - `TODO`: TODO.
+
+#![feature(array_methods)]
+#![cfg_attr(not(feature = "std"), no_std)]
+#![recursion_limit = "256"]
+
+use codec::{Decode, Encode};
+use frame_support::dispatch::Weight;
+use frame_support::{
+    decl_error, decl_event, decl_module, decl_storage,
+    dispatch::DispatchResult,
+    ensure,
+    traits::{Currency, ExistenceRequirement},
+    unsigned::TransactionSource,
+    unsigned::TransactionValidity,
+};
+use frame_system::{ensure_none, RawOrigin};
+use pallet_staking::{self as staking, RewardDestination};
+use polymesh_common_utilities::{
+    constants::{currency::POLY, REWARDS_MODULE_ID},
+    traits::identity::Config as IdentityConfig,
+    with_transaction,
+};
+use polymesh_primitives::Balance;
+use sp_runtime::transaction_validity::{InvalidTransaction, TransactionLongevity};
+use sp_runtime::{
+    traits::{AccountIdConversion, StaticLookup, Verify},
+    transaction_validity::ValidTransaction,
+    DispatchError,
+};
+use sp_std::{convert::TryInto, prelude::*};
+
+#[cfg(feature = "runtime-benchmarks")]
+pub mod benchmarking;
+
+type Staking<T> = staking::Module<T>;
+type BalanceOf<T> = <<T as pallet_staking::Config>::Currency as Currency<
+    <T as frame_system::Config>::AccountId,
+>>::Balance;
+
+pub trait Config: frame_system::Config + IdentityConfig + staking::Config {
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+    /// Weight information for extrinsic of the sto pallet.
+    type WeightInfo: WeightInfo;
+}
+
+pub trait WeightInfo {
+    fn claim_itn_reward() -> Weight;
+}
+
+/// Represents an Itn reward's status.
+#[derive(Decode, Encode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ItnRewardStatus {
+    Unclaimed(Balance),
+    Claimed,
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// Address was not found in the list of Itn addresses.
+        UnknownItnAddress,
+        /// Itn reward was already claimed.
+        ItnRewardAlreadyClaimed,
+        /// Provided signature was invalid.
+        InvalidSignature,
+        /// Balance can not be converted to a primitive.
+        UnableToCovertBalance,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Rewards {
+        /// Map of (Itn Address `AccountId`) -> (Reward `ItnRewardStatus`).
+        pub ItnRewards get(fn itn_rewards): map hasher(blake2_128_concat) T::AccountId
+            => Option<ItnRewardStatus>;
+    }
+    add_extra_genesis {
+        config(itn_rewards): Vec<(T::AccountId, Balance)>;
+        build(|config: &GenesisConfig<T>| {
+            for (account, balance) in &config.itn_rewards {
+                <ItnRewards<T>>::insert(account, ItnRewardStatus::Unclaimed(*balance));
```

Comment:
Let's move this to a special `register_itn_reward` extrinsic, called here, that uses `ensure_root`

Reply:
I will do this in a new PR, along with other additional extrinsics.